ocular/src/chain_client.rs

```diff
@@ -21,7 +23,9 @@ pub struct ChainClient {
 }
 
 impl ChainClient {
-    pub fn new(config: ChainClientConfig) -> Result<Self, ChainClientError> {
+    pub fn new(chain_name: &str) -> Result<Self, ChainClientError> {
+        let chain = executor::block_on(async { get_chain(chain_name).await })?;
```

Comment:
If we're blocking here isn't it redundant to make it an async call? (completely new to rust threading so that's an open question rather than something to definitively change)

Reply:
`get_chain` is an async method but the constructor isn't. In order to await the chain info without firing the thread and forgetting it has to block.